pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -526,7 +526,7 @@ protected void handleConnect(CommandConnect connect) {
             // Not find provider named authMethod. Most used for tests.
             // In AuthenticationDisabled, it will set authMethod "none".
             if (authenticationProvider == null) {
-                authRole = getBrokerService().getAuthenticationService().getAnonymousUserRole();
+                authRole = getBrokerService().getAuthenticationService().getAnonymousUserRole().get();
```

Comment:
getOrElseThrow(() -> AuthenticationException("No anonymous role, and no authentication provider configured"))